mlir/test/Conversion/TosaToLinalg/tosa-to-linalg-resize.mlir

```diff
@@ -38,71 +50,22 @@ func.func @unary_resize_bilinear_fp16(%arg0 : tensor<3x1x1x7xf16>) -> tensor<3x1
 
 // CHECK-LABEL: @unary_resize_nearest_i8
 func.func @unary_resize_nearest_i8(%arg0 : tensor<3x1x1x7xi8>) -> tensor<3x1x1x7xi8> {
-  %resize = "tosa.resize"(%arg0) {mode = "NEAREST_NEIGHBOR", scale = array<i64: 2, 1, 3, 1>, offset = array<i64: 0, 0>, border = array<i64: 0, 0>} : (tensor<3x1x1x7xi8>) -> tensor<3x1x1x7xi8>
+  %scale = tosa.const_shape { value = dense<[2, 1, 3, 1]> : tensor<4xindex> } : () -> !tosa.shape<4>
+  %offset = tosa.const_shape { value = dense<0> : tensor<2xindex> } : () -> !tosa.shape<2>
+  %border = tosa.const_shape { value = dense<0> : tensor<2xindex> } : () -> !tosa.shape<2>
+  %resize = tosa.resize %arg0, %scale, %offset, %border {mode = "NEAREST_NEIGHBOR"} : (tensor<3x1x1x7xi8>, !tosa.shape<4>, !tosa.shape<2>, !tosa.shape<2>) -> tensor<3x1x1x7xi8>
   // CHECK: return %arg0
   return %resize : tensor<3x1x1x7xi8>
 }
 
 // -----
 
-// CHECK-LABEL: @broadcast_resize_nearest_f32
-func.func @broadcast_resize_nearest_f32(%arg0 : tensor<3x1x1x7xf32>) -> tensor<3x1x5x7xf32> {
```

Comment:
Since we took the decision to avoid removing `MaterializeResizeBroadcast` for now, we should keep these tests in place. These will likely fail until an exception for this broadcasting is added to https://github.com/llvm/llvm-project/pull/124956/files#diff-90956ba24a2a97cc56a9a3659c7e46e56f1bd791a869246c6a758f9c93f1434fR1762

Reply:
Added back these test cases and check the expected error messages.

Reply:
Apologies, there might have been some confusion here. If the intention of keeping `MaterializeResizeBroadcast` was to avoid breaking things for other users that potentially rely on the broadcasting functionality, I think we’d need to make sure the tests that have been replaced don’t have any changes to them (in particular aren’t checking for an expected error). Currently someone that relies on this broadcasting functionality will hit the same error which may break existing flows.

My thinking was to make an exception in the verifier so that it doesn’t check the output shape when broadcasting is involved. We can then repeat this check in the validation pass without the exception to properly check we conform to the specification

Reply:
Thanks for your clarification. It makes sense to me.
I loose the checking in ResizeOp::verify() and add checking to TosaValidation pass.

Reply:
Thanks for the updates - this looks good to me now!